ui/src/components/RuleCard/index.jsx

```diff
@@ -211,6 +214,97 @@ function RuleCard({
   const priorityTitle = isScheduledPriorityUpdate
     ? 'Scheduled Priority'
     : 'Priority';
+  const handleTooltipOpen = () => {
+    setOpen(true);
+    setSeeMore('...see less');
+  };
+
+  const handleTooltipClose = () => {
+    setOpen(false);
+    setSeeMore('...see more');
+  };
+
+  const splitAmount = 5;
+  const isOsVersionLong = () => {
+    if (rule.osVersion) {
+      const osVersionsLength = rule.osVersion.split(',').length;
+
+      if (osVersionsLength > splitAmount) {
+        return true;
+      }
+
+      return false;
+    }
+  };
+
+  // Function to get index of occurrence of (,)
+  function getIndex(word, substr, occur) {
+    let occurrence = occur;
+    const Len = word.length;
+    let i = -1;
+
+    // eslint-disable-next-line no-plusplus
+    while (occurrence-- && i++ < Len) {
```

Comment:
**suggestion**: Consider handling the case where `substring` isn't found (for re-use.)

Reply:
**question:** Is it necessary to use unary operators in this code block?

Reply:
**suggestion:** To adhere to the eslint rule and improve code readability, consider changing the `while` loop:
```javascript
  while (occurrence > 0 && i < word.length) {
    i = word.indexOf(substr, i + 1);

    if (i < 0) break;

    occurrence -= 1;
  }
```

Reply:
**issue:** The code, while clever, could be more self-explanatory.

Reply:
**suggestion:** Rename occurrence to `remainingOccurrences` and `occur` to `occurrences` to better reflect their purpose in the code.

Reply:
**suggestion:** Consider using `String.prototype.indexOf()`

```javascript
function getIndex(word, substr, occur) {
  let currentIndex = -1;

  for (let count = 0; count < occur && currentIndex < word.length; count++) {
    currentIndex = word.indexOf(substr, currentIndex + 1);

    if (currentIndex === -1) {
      break;
    }
  }

  return currentIndex;
}
```

Reply:
**todo:** I would like a unit test for this functionality in case we have to change it in the future (or we want to re-use the `getIndex` function.) You could move the function into a logic module like [elementsHeight](https://github.com/mozilla-releng/balrog/blob/main/ui/src/utils/elementsHeight.js). Then add a unit test in [ui/test](https://github.com/mozilla-releng/balrog/blob/main/ui/test). That will run as part of the [application's CI process](https://github.com/mozilla-releng/balrog/pull/3033/checks?check_run_id=18011913985). I will open an issue to add this test suite once these changes are merged.

Reply:
**tip:** You can run tests locally using docker:
```
(balrog) ➜  balrog git:(highlight-partial-match-releases) docker exec -it balrog-balrogui-1 yarn test
yarn run v1.22.19
$ jest
 PASS  test/ListRules_test.jsx

Test Suites: 1 passed, 1 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        0.506 s, estimated 1 s
Ran all test suites.
Done in 0.98s.
```

Reply:
I will open an issue to add this test suite once these changes are merged :)